### PR TITLE
Rename extension actions to use mod/mod component terminology

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Rename `useActivateRecipe` -> `useActivateMod`
- Rename actions in extensions slice
- Refactoring via IntelliJ / changing the name and fixing compiler errors

## Remaining Work

- [x] Fix uninstallUtils mock
- [x] Fix broken test `ConvertToModModal` test — somehow the type checker wasn't flagging a mistake 🤷 

## Future Work

- The uninstallUtils files is currently mocked as a nop in `src/__mocks__/@/store/uninstallUtils.ts.` We probably shouldn't be mocking the file there. (Tests should run the actual reducer code, and the platform API should be the mocking boundary)
- Clarify "remove" vs. "uninstall" vs. "deactivate" terminology. Should likely be called "deactivate". But in some places we do need to clarify deactivate vs. also clearing mods from existing tabs/frames

## Checklist

- [x] Add jest or playwright tests and/or storybook stories: N/A
- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/store/extensionsSlice.ts [hunk 1]
```diff
@@ -81,37 +81,44 @@ const extensionsSlice = createSlice({
 
     activateStandaloneModDefinition(
       state,
-      { payload }: PayloadAction<{ extension: StandaloneModDefinition }>,
+      {
+        payload,
+      }: PayloadAction<{ standaloneModDefinition: StandaloneModDefinition }>,
```
Comment (twschiller) [inline src/store/extensionsSlice.ts:86]: Might consider passing directly as payload vs. passing object given that the type is unambiguous. This method is likely deprecated though to be removed once we drop standalone mod activation

File: src/store/uninstallUtils.ts [hunk 2]
```diff
@@ -21,58 +21,57 @@ import {
   removeDraftModComponentsForMod,
 } from "@/store/editorStorage";
 import { actions as extensionActions } from "@/store/extensionsSlice";
-import { removeExtensionForEveryTab } from "@/background/messenger/api";
+import { removeModComponentForEveryTab } from "@/background/messenger/api";
 import { uniq } from "lodash";
 import { type UnresolvedModComponent } from "@/types/modComponentTypes";
 import { type RegistryId } from "@/types/registryTypes";
 import { type UUID } from "@/types/stringTypes";
```
Comment (twschiller) [inline src/store/uninstallUtils.ts:28]: This file mixes uninstall/remove terminology

File: src/store/extensionsSlice.ts
Comment (twschiller) [inline src/store/extensionsSlice.ts:1]: Eventually we'll want to determine if convention for using object payloads or not. E.g., single argument actions are more concise without object prop names and the meaning of the method argument is generally clear

**Comment by grahamlangford** [comment]
NIT: Shouldn't `uninstall` be `deactivate`?

**Comment by twschiller** [comment]
> NIT: Shouldn't `uninstall` be `deactivate`?

Had called out in Future Work:  Clarify "remove" vs. "uninstall" vs. "deactivate" terminology. Should likely be called "deactivate". But in some places we do need to clarify deactivate vs. also clearing mods from existing tabs/frames

**Comment by grahamlangford** [comment]
> > NIT: Shouldn't `uninstall` be `deactivate`?
> 
> Had called out in Future Work: Clarify "remove" vs. "uninstall" vs. "deactivate" terminology. Should likely be called "deactivate". But in some places we do need to clarify deactivate vs. also clearing mods from existing tabs/frames

I missed that. 👍

**Review by grahamlangford — APPROVED** [review]

File: src/store/extensionsSlice.ts
Comment (grahamlangford) [reply src/store/extensionsSlice.ts:1]: In a vanilla JavaScript app, I would push for object payloads even with single arguments, but with TypeScript, I don't think they're necessary for single args

**Pull request merged**